shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformChannel.java

```diff
@@ -4,26 +4,536 @@
+     */
+    void vibrateHapticFeedback(@NonNull HapticFeedbackType feedbackType);
+
+    /**
+     * The Flutter application would like to display in the given {@code androidOrientation}.
+     */
+    // TODO(mattcarroll): add @ScreenOrientation annotation
+    void setPreferredOrientations(int androidOrientation);
+
+    /**
+     * The Flutter application would like to be displayed in Android's app switcher with
+     * the visual representation described in the given {@code description}.
+     * <p>
+     * See the related Android documentation:
+     * https://developer.android.com/guide/components/activities/recents
+     */
+    void setApplicationSwitcherDescription(@NonNull AppSwitcherDescription description);
+
+    /**
+     * The Flutter application would like the Android system to display the given
+     * {@code overlays}.
+     * <p>
+     * {@link SystemUiOverlay#TOP_OVERLAYS} refers to system overlays such as the
+     * status bar, while {@link SystemUiOverlay#BOTTOM_OVERLAYS} refers to system
+     * overlays such as the back/home/recents navigation on the bottom of the screen.
+     * <p>
+     * An empty list of {@code overlays} should hide all system overlays.
+     */
+    void showSystemOverlays(@NonNull List<SystemUiOverlay> overlays);
+
+    /**
+     * The Flutter application would like to restore the visibility of system
+     * overlays to the last set of overlays sent via {@link #showSystemOverlays(List)}.
+     * <p>
+     * If {@link #showSystemOverlays(List)} has yet to be called, then a default
+     * system overlay appearance is desired:
+     * <p>
+     * {@code
+     * View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+     * }
+     */
+    void restoreSystemUiOverlays();
+
+    /**
+     * The Flutter application would like the system chrome to present itself with
+     * the given {@code systemUiOverlayStyle}, i.e., the given status bar and
+     * navigation bar colors and brightness.
+     */
+    void setSystemUiOverlayStyle(@NonNull SystemChromeStyle systemUiOverlayStyle);
+
+    /**
+     * The Flutter application would like to pop the top item off of the Android
+     * app's navigation back stack.
+     */
+    void popSystemNavigator();
+
+    /**
+     * The Flutter application would like to receive the current data in the
+     * clipboard and have it returned in the given {@code format}.
+     */
+    @Nullable
+    CharSequence getClipboardData(@Nullable ClipboardContentFormat format);
+
+    /**
+     * The Flutter application would like to set the current data in the
+     * clipboard to the given {@code text}.
+     */
+    void setClipboardData(@NonNull String text);
+  }
+
+  /**
+   * Types of sounds the Android OS can play on behalf of an application.
+   */
+  public enum SoundType {
+    CLICK("SoundType.click");
+
+    static SoundType fromValue(@NonNull String encodedName) {
```

Comment:
The public members in this and the related enums also probably need Javadoc.

Reply:
I'm going to opt to wait until someone asks questions about these enums to give any javadocs because to me they seem self-explanatory.